### PR TITLE
test: fix tests in dart 2.19

### PR DESCRIPTION
**Pull request by amondnet**
Describe the changes proposed in this Pull Request.

If the PR fixes a specific issue, reference the issue with **`Fixes #`**.

Dart 2.19 doesn't delegate inaccessible private names to noSuchMethod.

https://github.com/dart-lang/sdk/blob/main/CHANGELOG.md#2190
https://github.com/dart-lang/language/issues/2020
https://github.com/dart-lang/sdk/issues/49687
https://github.com/dart-lang/site-www/issues/4263

https://github.com/mobxjs/mobx.dart/blob/master/mobx/test/observable_test.dart

The test fails on dart 2.19.

https://github.com/mobxjs/mobx.dart/actions/runs/3262020308/jobs/5359681093

---
## Pull Request Checklist

- [x] If the changes are being made to code, ensure the **version in `pubspec.yaml`** is updated. 
- [x] Increment the **`major`/`minor`/`patch`/`patch-count`**, depending on the complexity of change
- [x] Add the necessary **unit tests** to ensure the coverage does not drop
- [x] Update the **Changelog** to include all changes made in this PR, organized by version
- [x] Run the **`melo run set_version` command** from the root directory
- [ ] Include the **necessary reviewers** for the PR
- [x] Update the docs if there are any API changes or additions to functionality

### CODE REVIEW

**Review by fzyzcjy — APPROVED** [review]
LGTM except the codecov

**Comment by tim-eucalyptus** [comment]
Would be great to see this merged as it's holding up: <https://github.com/mobxjs/mobx.dart/pull/877>. Thanks!

File: mobx/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ## 2.1.1
 
 - Allow a custom equals parameter for ObservableStream - [@amondnet](https://github.com/amondnet)
+- Fix tests in dart 2.19 - [@amondnet](https://github.com/amondnet)
```
Comment (pavanpodila) [inline mobx/CHANGELOG.md:4]: add a new version for this

File: mobx/lib/src/core/reaction.dart [hunk 2]
```diff
@@ -64,7 +64,7 @@ class ReactionImpl implements Reaction {
     _isRunning = false;
 
     if (_isDisposed) {
-      _context._clearObservables(this);
+      _context.clearObservables(this);
```
Comment (pavanpodila) [inline mobx/lib/src/core/reaction.dart:67]: can we cover this use case with a test ?

**Review by pavanpodila — CHANGES_REQUESTED** [review]

File: mobx/CHANGELOG.md (hunk 1 above)
Comment (amondnet) [reply mobx/CHANGELOG.md]: @pavanpodila done

File: mobx/lib/src/core/reaction.dart (hunk 2 above)
Comment (amondnet) [reply mobx/lib/src/core/reaction.dart:67]: @pavanpodila done

**Comment by tim-eucalyptus** [comment]
Any update on this??

**Comment by HudsonAfonso** [comment]
news?

**Pull request merged**